Sources/StreamChat/ChatClient.swift

```diff
@@ -342,6 +349,42 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
             waiters.removeAll()
         }
     }
+    
+    private func handleAppDidEnterBackground() {
+        guard connectionStatus == .connected else { return }
+        guard config.staysConnectedInBackground else {
+            // We immediately disconnect
+            clientUpdater.disconnect(source: .systemInitiated)
+            return
+        }
+        guard let scheduler = backgroundTaskScheduler else { return }
+        
+        let succeed = scheduler.beginTask { [weak self] in
+            self?.clientUpdater.disconnect(source: .systemInitiated)
+            // We need to call `endBackgroundTask` else our app will be killed
+            self?.cancelBackgroundTaskIfNeeded()
+        }
+        
+        if !succeed {
+            // Can't initiate a background task, close the connection
+            clientUpdater.disconnect(source: .systemInitiated)
+        }
+    }
+    
+    private func handleAppDidBecomeActive() {
+        cancelBackgroundTaskIfNeeded()
+        
+        guard config.shouldConnectAutomatically else { return }
+        guard connectionStatus != .connected && connectionStatus != .connected else {
```

Comment:
nit: we should maybe add one more `&& connectionStatus != .connected` to be 300% sure :trollface:

Reply:
```swift
let confidenceChecks = (1...10).random()
for _ in confidenceChecks {
  guard connectionStatus != .connected else { return }
}
```